diff -ruN .venv/lib/python3.12/site-packages/langchain_core/vectorstores/base.py ./custom_langchain_core/vectorstores/base.py
--- .venv/lib/python3.12/site-packages/langchain_core/vectorstores/base.py	2025-02-22 16:26:38
+++ ./custom_langchain_core/vectorstores/base.py	2025-02-22 16:26:11
@@ -501,6 +501,12 @@
         """
         relevance_score_fn = self._select_relevance_score_fn()
         docs_and_scores = self.similarity_search_with_score(query, k, **kwargs)
+        print("docs_and_scores")
+        print(docs_and_scores)
+        for doc, score in docs_and_scores:
+            doc.metadata["similarity_score"] = score
+            doc.metadata["relevance_score"] = relevance_score_fn(
+                score)
         return [(doc, relevance_score_fn(score)) for doc, score in docs_and_scores]
 
     async def _asimilarity_search_with_relevance_scores(
